JDBC/OAuthSampleApp/src/main/java/OAuthSampleApp.java

```diff
@@ -10,91 +10,310 @@
+		try {
+			String postOpts = getEncodedParamsString(formData);
+			byte[] postData = postOpts.getBytes("UTF-8");
+			int postDataLength = postData.length;
+			URL url = new URL(prop.getProperty("TokenUrl"));
+			HttpURLConnection connection = (HttpURLConnection) url.openConnection();
+			try {
+				connection.setDoOutput(true);
+				connection.setUseCaches(false);
+				connection.setRequestMethod("POST");
+				connection.setRequestProperty("Content-Type", "application/x-www-form-urlencoded");
+				connection.setRequestProperty("Content-Length", Integer.toString(postDataLength));
+				connection.setRequestProperty("Accept", "application/json");
+				connection.getOutputStream().write(postData);
+				BufferedInputStream in = new BufferedInputStream(connection.getInputStream());
+				ByteArrayOutputStream buf = new ByteArrayOutputStream();
+				readResult(in, buf);
+				String res = buf.toString("UTF-8");
+				JsonElement jElement = JsonParser.parseString(res);
+				JsonObject jObject = jElement.getAsJsonObject();
+				// set Tokens as System Properties - new values to access_token and
+				// refresh_token
+				String accessToken = jObject.has("access_token") ? jObject.get("access_token").getAsString() : null;
+				String refreshToken = jObject.has("refresh_token") ? jObject.get("refresh_token").getAsString() : null;
+				if (null == accessToken || null == refreshToken) {
+					throw new Exception(
+							"Access/refresh token is null, Please verify the config.properties for proper inputs.");
+				}
+				System.setProperty(OAUTH_ACCESS_TOKEN_VAR_STRING, accessToken);
+				System.setProperty(OAUTH_REFRESH_TOKEN_VAR_STRING, refreshToken);
+			} catch (UnsupportedEncodingException uee) {
+				uee.printStackTrace();
+			} catch (Exception e) {
+				String res = "";
+				try {
+					BufferedInputStream in = new BufferedInputStream(connection.getErrorStream());
+					ByteArrayOutputStream buf = new ByteArrayOutputStream();
+					readResult(in, buf);
+					res = buf.toString("UTF-8");
+				} catch (Exception ex) {
+					// Skip when this happens, but print the error.
+					ex.printStackTrace();
+				}
+				throw e;
+			} finally {
+				connection.disconnect();
+			}
+		} catch (IOException unreportedex) {
+			unreportedex.printStackTrace();
+		}
+	}
+
+	// If there is no access token, Get it using password grant
+	private static void EnsureAccessToken() throws Exception {
+		try {
+			String accessToken = System.getenv(OAUTH_ACCESS_TOKEN_VAR_STRING);
+			String refreshToken = System.getenv(OAUTH_REFRESH_TOKEN_VAR_STRING);
+			if (null == accessToken || accessToken.isEmpty()) {
+				if (null == refreshToken || refreshToken.isEmpty()) {
+					// Obtain first access token and refresh Tokens
+					System.out.println("Access token is invalid/expired, trying to do refresh");
+					GetTokensByPasswordGrant();
+				} else {
+					try{
+						System.out.println("Attempting to use given refresh token.");
+						GetTokensByRefreshGrant();
+					}catch(){
```

Comment:
This catch is empty

Reply:
Added exception variable for catch block